<!--
 * @Author: wangkaifei kfwang@stu.xidian.edu.cn
 * @Date: 2025-05-11 14:37:59
 * @LastEditors: wangkaifei kfwang@stu.xidian.edu.cn
 * @LastEditTime: 2025-05-11 19:56:16
 * @FilePath: \public\output.md
 * @Description: è¿™æ˜¯é»˜è®¤è®¾ç½®,è¯·è®¾ç½®`customMade`, æ‰“å¼€koroFileHeaderæŸ¥çœ‹é…ç½® è¿›è¡Œè®¾ç½®: https://github.com/OBKoro1/koro1FileHeader/wiki/%E9%85%8D%E7%BD%AE
-->
## ğŸ› ï¸ Output Tree


<pre>output-root/
â”œâ”€â”€ novel.spectra # Identified peptide-spectrum matches (PSMs) corresponding to novel peptides
â”œâ”€â”€ annoed.spectra # Identified PSMs corresponding to known peptides (from reference database)
â”œâ”€â”€ stage2_seqs.spectra # Database search results filtered by grouped FDR
â”œâ”€â”€ target_GSSP.txt # Identified novel peptides (each line is an amino acid sequence)
â”œâ”€â”€ dna_anno.panno # Complete CDS annotation results inferred from novel peptides
â”œâ”€â”€ dna_anno-filtered.panno # High-confidence CDS filtering by Multi-stage CDS mapping group-based filter (complex species only)
â”œâ”€â”€ part1.panno # Part 1 results: outputs by CDS mapping group unit, representing the complete gene coding regions (complex species only)
â”œâ”€â”€ part1.gff # CDS classified in Part 1, output in GFF format (complex species only)
â”œâ”€â”€ part2.panno # Part 2 results: CDS inference results supported by genome, transcriptome, and proteome, but not passing the Multi-stage CDS mapping group-based filter (complex species only)
â”œâ”€â”€ part2.gff # CDS classified in Part 2, output in GFF format (complex species only)
â”œâ”€â”€ part3.panno # Part 3 results: CDS consists of RNA-exclusive CDSs lacking DNA concordance (complex species only)
â”œâ”€â”€ part4.panno # Part 4 results: CDS with novel peptides but lacking RNA evidence (complex species only)
â”œâ”€â”€ part4.gff # CDS classified in Part 4, output in GFF format (complex species only)
â”œâ”€â”€ pFind_re2/ # Directory storing results of Step 1 in database search stage
â”‚ â”œâ”€â”€ pFind-Filtered.spectra # PSM identification results of Step 1
â”‚ â”œâ”€â”€ MSdel/ # Directory storing spectra not interpreted in Step 1
â”‚ â”‚ â””â”€â”€ XXX.mgf          
â”œâ”€â”€ pFind_res2/ # Directory storing result files of Step 2 in database search stage
â”‚ â”œâ”€â”€ DBReducer/ # DBReducer results for processing the customized database
â”‚ â”‚ â”‚â”€â”€ Reduced.fasta # Result of reduced customized database
â”‚ â”‚ â”‚â”€â”€ DNARed/ # Directory storing DBReducer results for DNA six-frame translated database
â”‚ â”‚ â””â”€â”€ RNARed/ # Directory storing DBReducer results for RNA three-frame translated database
â”‚ â”œâ”€â”€ pFind-Filtered.spectra # PSM identification results of Step 2 </pre>

### Please prioritize the following result files:
- novel.spectra
- [dna_anno.panno](#dna_anno.panno)
#### For complex species, the following result files are also recommended:
- [part1.panno](#part1.panno)
- [dna_anno-filtered.panno](#dna_anno.panno)
  
## ğŸ› ï¸ Output Documentation
### dna_anno.panno
|Column               |   Content                                                         |
|--------------------|----------------------------------------------------------------------|
| `Is_rna`          | 0 or 1. 1 indicates RNA-originated only, 0 indicates inferred from genome.|
| `Is_mutation`          | 0 or 1. 1 indicates this CDS contains mutations |
| `Is_splice`      | 0 or 1. 1 indicates a splice site with peptide evidence                       |
| `Is_subset`       | 0 or 1. 1 indicates the novel peptides supporting this CDS are a true subset of those from another CDS. |
| `Chr`        | Chromosome name. For example, DNA.fna_line6152619_NC_000003.12 indicates chromosome NC_000003.12 located on line 6152619 of the DNA.fna file.|
| `ID`       | æœ¬CDSå¯¹åº”åˆ°çš„å®šåˆ¶æ•°æ®åº“ä¸­çš„protein IDã€‚                          |
| `Frame`     | é˜…è¯»æ¡†æ¥æºã€‚å› ä¸ºpAnnoæ”¯æŒä¸åŒframeæ‹¼æ¥çš„å‰ªæ¥ä½“ï¼Œæ‰€ä»¥æ­¤å¤„ç”¨'4'æ ‡è¡¨ç¤ºå‰ªæ¥ä½“ |
| `pep_q_value`        | æ”¯æŒæœ¬CDSçš„novel peptidesçš„æœ€å°q-value                                      |
| `Novel_pep`      | æ”¯æŒæœ¬CDSçš„novel peptidesï¼Œä»¥';'åˆ†éš”                                   |
| `Old_pep`         | æœ¬CDSè¦†ç›–åˆ°çš„known peptides (derived from reference database)ï¼Œä»¥';'åˆ†éš” |
| `Start_codon`        | æ¨æ–­çš„èµ·å§‹å­ã€‚è¯·æ³¨æ„ï¼Œæ­¤é¡¹ä»…å¯¹äºç®€å•ç‰©ç§(å‡ ä¹æ— å‰ªæ¥)æœ‰æ•ˆï¼Œå¤æ‚ç‰©ç§ï¼ˆå…·æœ‰å¤§é‡å‰ªæ¥ï¼‰è¯·å‚è€ƒpart1.pannoæ–‡ä»¶|
| `Start`       | æ•´æ•°ï¼ŒCDSåœ¨æœ¬æŸ“è‰²ä½“ä¸Šå¼€å§‹ç¿»è¯‘çš„ä½ç½® ï¼ˆæ­¤é¡¹ä»…å¯¹äºç®€å•ç‰©ç§æœ‰æ•ˆï¼Œå¤æ‚ç‰©ç§è¯·å‚è€ƒpart1.pannoæ–‡ä»¶ï¼‰ |
| `End`    | æ•´æ•°ï¼ŒCDSåœ¨æœ¬æŸ“è‰²ä½“ä¸Šç»“æŸç¿»è¯‘çš„ä½ç½® ï¼ˆæ­¤é¡¹ä»…å¯¹äºç®€å•ç‰©ç§æœ‰æ•ˆï¼Œå¤æ‚ç‰©ç§è¯·å‚è€ƒpart1.pannoæ–‡ä»¶ï¼‰|
| `Strand`  | æ­£è´Ÿé“¾æ ‡è®°                              |
| `Mutation_info` | çªå˜ä¿¡æ¯ã€‚1:5:S->Tè¡¨ç¤º`Novel_pep`ä¸­çš„ç¬¬ä¸€æ¡è‚½æ®µçš„ç¬¬5ä¸ªæ°¨åŸºé…¸ä»Sçªå˜ä¸ºäº†Tï¼Œä»¥';'åˆ†éš”|
| `Splice_info`         | å‰ªæ¥ä¿¡æ¯ã€‚"4683787;5174351"è¡¨ç¤º4683787å’Œ5174351ä½ç½®çš„ç¢±åŸºåˆ†åˆ«ä¸ºdonorå’Œacceptorä½ç‚¹ |
| `Anno_pro`      | å¯¹åŸºå› ç»„æ³¨é‡Šçš„ä¿®æ­£ã€‚"Start:5172647,End:5172740"è¡¨ç¤ºå¯¹æœ¬DNAé“¾çš„(5172647,5172740)æ®µçš„ç¼–ç åŒºæ³¨é‡Šåšäº†ä¿®æ­£ã€‚ä»¥';'åˆ†éš”                                      |
| `Pro_seq` | CDSå¯¹åº”çš„æ°¨åŸºé…¸åºåˆ—                      |
| `Gene_seq` | CDSå¯¹åº”çš„ç¢±åŸºåºåˆ—                                |
| `Gene_q_value`  | æš‚æœªå¯ç”¨ã€‚                             |
| `Target/Decoy`  | æš‚æœªå¯ç”¨ã€‚                             |

### part1.panno
æœ¬æ–‡ä»¶å†…å®¹ä¸dna_anno.pannoåŸºæœ¬ä¸€è‡´ï¼Œä½†å…·æœ‰äºŒçº§æ ‡é¢˜ã€‚æ¯ä¸ªitemä»£è¡¨ä¸€ä¸ªCDS mapping gropï¼Œè¿™äº›CDSè¢«æ¨æ–­ä¸ºç»„æˆä¸€ä¸ªå®Œæ•´çš„åŸºå› ç¼–ç åŒºã€‚
itemçš„ç¬¬ä¸€è¡Œæ ‡æ³¨äº†è¯¥åŸºå› ä¸­è·¨è¶Šå…¨éƒ¨è¡¨è¾¾åŒºåŸŸçš„éƒ¨åˆ†ï¼ˆåŒ…æ‹¬å†…éƒ¨çš„éç¼–ç åŒºï¼‰ï¼Œéœ€è¦æ³¨æ„:
- `Is_rna` ä¸€åˆ—çš„å€¼ä¸º1ï¼Œè¡¨ç¤ºCDS mapping groupçš„å¼€å§‹
- `Splice_info`ä¸€åˆ—æœ‰`n-1`å¯¹donorå’Œacceptorä½ç‚¹ï¼Œ`n`è¡¨ç¤ºç»„ä¸­çš„CDSæ•°é‡ã€‚æ¯”å¦‚`(62535943,62535191);(62535000,62534074);`è¡¨ç¤ºCDS mapping groupä¸­æœ‰3ä¸ªCDSï¼Œåˆ†åˆ«åœ¨62535943bp-62535191bpå’Œ62535000bp-62534074bpå¤„å‘ç”Ÿå‰ªæ¥ã€‚
- `Pro_seq`è¡¨ç¤ºå®Œæ•´ç¼–ç åŒºçš„æ°¨åŸºé…¸åºåˆ—
- `Gene_seq`è¡¨ç¤ºå®Œæ•´ç¼–ç åŒºçš„ç¢±åŸºåºåˆ—
- `Mutation`æœ¬åˆ—ä½äºæœ«å°¾ã€‚è¡¨ç¤ºå®Œæ•´ç¼–ç åŒºä¸­çš„çªå˜ä¿¡æ¯ï¼Œ`102:Q->R`è¡¨ç¤º`Pro_seq`ä¸­çš„ç¬¬102ä¸ªæ°¨åŸºé…¸ä»Qçªå˜ä¸ºRã€‚ä»¥';'åˆ†éš”

itemç¬¬ä¸€è¡Œä¹‹åçš„æ¯è¡Œç¼©è¿›ä¸€åˆ—ï¼ŒæŒ‰äºŒçº§æ ‡é¢˜è¾“å‡ºï¼Œä»£è¡¨è¯¥ä¸€ä¸ªCDSï¼Œæ˜¯è¯¥åŸºå› çš„ä¸€ä¸ªéƒ¨åˆ†:
- `Start_codon`åªé’ˆå¯¹CDS mapping groupä¸­å¤„äºNç«¯çš„CDSåšèµ·å§‹å­æ¨æ–­ï¼Œå¹¶å°†ç»“æœå†™å…¥itemç¬¬ä¸€è¡Œçš„`Start_codon`åˆ—ä¸­ã€‚å…¶ä»–CDSçš„`Start_codon`åˆ—å‡ä¸ºç©ºã€‚
- `Start`å’Œ`End`åˆ—è¡¨ç¤ºè¯¥CDSåœ¨æœ¬æŸ“è‰²ä½“ä¸Šçš„ç¿»è¯‘èµ·å§‹å’Œç»“æŸä½ç½®